### PR TITLE
Bugfix/failing legacy tests

### PR DESCRIPTION
**Pull request by luke-hill**
## Summary

Fix up legacy tests. Also I've jotted down a load of new fixes I need to do to improve / future proof some of the tech debt in the testing suite. The issue came from rails auto-generator not inserting a datepicker with a certain expected format (>6 years of legacy dates).

For some reason the rake notes test fails for me locally but passes in CI. So possibly one to watch out for.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue).
- [ ] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).

## Checklist:

- [ ] I've added tests for my code.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: features/capybara_javascript_drivers.feature [hunk 1]
```diff
@@ -54,23 +54,27 @@ Feature: Capybara Javascript Drivers
       """
       @javascript
       Feature: Create appointments
-        Scenario: Visit the Constitution on May 17
+        Scenario: Create appointment using DB injection
           Given a random appointment
           And I am viewing the appointment
-          Then I should see "Random appointment"
+          Then I should see the correct appointment details
```
Comment (mvz) [inline features/capybara_javascript_drivers.feature:60]: :heart:

File: features/step_definitions/cucumber_rails_steps.rb [hunk 2]
```diff
@@ -15,12 +15,6 @@
   install_cucumber_rails :no_database_cleaner, :database_cleaner_active_record
 end
 
-Given('I have created a new Rails app and installed cucumber-rails without database_cleaner') do
-  rails_new
-  install_cucumber_rails :no_database_cleaner
-  overwrite_file('features/support/env.rb', "require 'cucumber/rails'\n")
-end
-
```
Comment (mvz) [inline features/step_definitions/cucumber_rails_steps.rb:23]: I think this step is still used in `features/capybara_javascript_drivers.feature` and `features/database_cleaner.feature`.

**Review by mvz — COMMENTED** [review]
Vary happy to see the test failures being fixed!

**Comment by olleolleolle** [comment]
Thanks so much for these, @luke-hill !

File: features/step_definitions/cucumber_rails_steps.rb (hunk 2 above)
Comment (luke-hill) [reply features/step_definitions/cucumber_rails_steps.rb:23]: I have no idea why this is deleted. I didn't even delete it myself 👻

**Review by olleolleolle — APPROVED** [review]
Yes!

:hearts: :yellow_heart: :green_heart: :blue_heart: :purple_heart:

**Comment by orien** [comment]
@luke-hill I've followed your lead and fixed a couple of failing tests. Perhaps you'll find commit af590eb helpful on this branch.

File: features/capybara_javascript_drivers.feature [hunk 3]
```diff
@@ -54,23 +54,27 @@ Feature: Capybara Javascript Drivers
...
       """
     And I write to "features/step_definitions/create_appointment_steps.rb" with:
       """
       Given('a random appointment') do
-        @appointment = Appointment.create!(name: 'Random appointment', when: DateTime.now)
+        @when = DateTime.now
+        @name = 'Random appointment for Cucumber Trainee'
+        @appointment = Appointment.create!(name: name, when: when)
```
Comment (mvz) [inline features/capybara_javascript_drivers.feature:67]: @luke-hill  There's a syntax error here.
```suggestion
        @appointment = Appointment.create!(name: @name, when: @when)
```

**Pull request merged**